lib/matplotlib/axes/_base.py

```diff
@@ -2103,22 +2104,31 @@ def autoscale(self, enable=True, axis='both', tight=None):
             The *tight* setting is retained for future autoscaling
             until it is explicitly changed.
 
+        This returns a context-manager/decorator that allows temporarily
+        setting the autoscale status, i.e.::
 
-        Returns None.
+            with axes.autoscale(enable): ...
+
+        will keep the autoscale status to `enable` for the duration of the
+        block only.
         """
-        if enable is None:
-            scalex = True
-            scaley = True
-        else:
-            scalex = False
-            scaley = False
+        orig_autoscale = self._autoscaleXon, self._autoscaleYon
+        if enable is not None:
             if axis in ['x', 'both']:
                 self._autoscaleXon = bool(enable)
-                scalex = self._autoscaleXon
             if axis in ['y', 'both']:
                 self._autoscaleYon = bool(enable)
-                scaley = self._autoscaleYon
-        self.autoscale_view(tight=tight, scalex=scalex, scaley=scaley)
+        self.autoscale_view(
```

Comment:
OK, so I misunderstood the code (I never used `autoscale(None)` so I had no idea what it did).  I still don't understand the actual behavior: autoscaling (in `autoscale_view`) is protected by the conditional `if scalex and self._autoscaleXon`, so I thought that just passing `scalex=True` would not perform an autoscale as long as `self._autoscaleXon` is `False`.
Where did I go wrong?